packages/fuchsia_remote_debug_protocol/test/src/runners/ssh_command_runner_test.dart

```diff
@@ -0,0 +1,142 @@
+import 'package:fuchsia_remote_debug_protocol/src/runners/ssh_command_runner.dart';
+
+void main() {
+  group('SshCommandRunner.constructors', () {
+    test('throws exception with invalid address', () async {
+      SshCommandRunner newCommandRunner() {
+        return new SshCommandRunner(address: 'sillyaddress.what');
+      }
+
+      expect(newCommandRunner, throwsArgumentError);
+    });
+
+    test('throws exception from injection constructor with invalid addr',
+        () async {
+      SshCommandRunner newCommandRunner() {
+        return new SshCommandRunner.withProcessManager(
+            const LocalProcessManager(),
+            address: '192.168.1.1.1');
+      }
+
+      expect(newCommandRunner, throwsArgumentError);
+    });
+  });
+
+  group('SshCommandRunner.run', () {
+    MockProcessManager mockProcessManager;
+    MockProcessResult mockProcessResult;
+    SshCommandRunner runner;
+
+    setUp(() {
+      mockProcessManager = new MockProcessManager();
+      mockProcessResult = new MockProcessResult();
+      when(mockProcessManager.run(typed(any))).thenReturn(mockProcessResult);
+    });
+
+    test('verify interface is appended to ipv6 address', () async {
+      final String ipV6Addr = 'fe80::8eae:4cff:fef4:9247';
+      final String interface = 'eno1';
+      runner = new SshCommandRunner.withProcessManager(
+        mockProcessManager,
+        address: ipV6Addr,
+        interface: interface,
+        sshConfigPath: '/whatever',
+      );
+      when<String>(mockProcessResult.stdout).thenReturn('somestuff');
+      when(mockProcessResult.exitCode).thenReturn(0);
+      await runner.run('ls /whatever');
+      final List<String> passedCommand =
+          verify(mockProcessManager.run(captureAny)).captured.single;
+      expect(passedCommand.indexOf('$ipV6Addr%$interface'), isNot(-1));
+    });
+
+    test('verify no percentage symbol is added when no ipv6 interface',
+        () async {
+      final String ipV6Addr = 'fe80::8eae:4cff:fef4:9247';
+      runner = new SshCommandRunner.withProcessManager(
+        mockProcessManager,
+        address: ipV6Addr,
+      );
+      when<String>(mockProcessResult.stdout).thenReturn('somestuff');
+      when(mockProcessResult.exitCode).thenReturn(0);
+      await runner.run('ls /whatever');
+      final List<String> passedCommand =
+          verify(mockProcessManager.run(captureAny)).captured.single;
+      expect(passedCommand.indexOf(ipV6Addr), isNot(-1));
+    });
+
+    test('verify commands are split into multiple lines', () async {
+      final String addr = '192.168.1.1';
+      runner = new SshCommandRunner.withProcessManager(mockProcessManager,
+          address: addr);
+      when<String>(mockProcessResult.stdout).thenReturn('''this
+          has
+          four
+          lines''');
+      when(mockProcessResult.exitCode).thenReturn(0);
+      final List<String> result = await runner.run('oihaw');
+      expect(result.length, 4);
```

Comment:
See also https://www.dartdocs.org/documentation/test/0.12.31/test/hasLength.html (when it fails, it produces a better error message)

Reply:
Done.